recipes/LJSpeech/TTS/vocoder/hifi_gan_unit/extract_code.py

```diff
@@ -9,21 +9,33 @@
 import logging
 import pathlib as pl
 
-import joblib
 import numpy as np
 import torch
 import torchaudio
 from tqdm import tqdm
 
 import speechbrain as sb
 from speechbrain.dataio.dataio import load_pkl, save_pkl
-from speechbrain.lobes.models.huggingface_transformers.wav2vec2 import Wav2Vec2
-
-OPT_FILE = "opt_ljspeech_extract.pkl"
+from speechbrain.lobes.models.huggingface_transformers import (
+    hubert,
+    wav2vec2,
+    wavlm,
+)
+from speechbrain.lobes.models.huggingface_transformers.discrete_ssl import (
+    DiscreteSSL,
+)
+
+OPT_FILE = "opt_ljspeech_extract_code.pkl"
 TRAIN_JSON = "train.json"
 VALID_JSON = "valid.json"
 TEST_JSON = "test.json"
 
+ENCODER_CLASSES = {
```

Comment:
"I prefer the way I'm doing it currently. The ssl_model is only used for token extraction and not during training, so I don't see why we need to instantiate the model in the YAML and it will require manual cleanup to free memory for HiFi-GAN training.

Reply:
i dont see what is the problem in defining it in yaml file instead of the python file... why ther isa difference for manual cleanup?